src/core_plugins/kibana/server/lib/supported_features/get_supported_features.js

```diff
@@ -0,0 +1,25 @@
+import { checkFieldStatsApi } from './feature_checks';
+
+
+const featureChecks = [
+  checkFieldStatsApi,
+];
+
+export async function getSupportedFeatures(callWithRequest) {
+  const { nodes: nodesInfo } = await callWithRequest('nodes.info', {
+    filterPath: [
+      'nodes.*.name',
+      'nodes.*.version',
+    ],
+  });
+
+  const clusterInfo = {
+    nodes: nodesInfo,
+  };
+
+  const featureFlags = featureChecks.reduce((featureFlags, featureCheck) => (
```

Comment:
Yes, I considered that. But in that scenario there would be no way for the feature checks to depend on each other and reuse each other's results. Or for one check to set multiple feature flags if they derive from the same source information.

I like the ability to see all available flags though. :thinking:

It probably is a mute argument right now as there is only one check and we don't know if there will ever be any others. :wink: